### ISSUE TITLE
Implement flyweight design pattern for ProfCardsCollection

### ISSUE THREAD

**Issue by mizu-benrb** [rainbow-cards/manoa-rainbow-cards#113]
Please do work for this task in a branch named issue-113. 

Currently, the ProfCardsCollection keeps track of duplicates of a card separately. Change the collection so it keeps count of total amount of cards in distribution BUT stores cards only once. Requires a different way of keeping tracking of ownership.
